editorModule.js

```diff
@@ -6,61 +6,60 @@ class EditorModule{
         this.addEventListeners();
     }
 
-    setupEditors(){
-        this.HTMLEditor = ace.edit("HTMLEditor");
-        this.CSSEditor = ace.edit("CSSEditor");
-        this.JSEditor = ace.edit("JSEditor");
-
-        this.HTMLEditor.setTheme("ace/theme/monokai");
-        this.CSSEditor.setTheme("ace/theme/monokai");
-        this.JSEditor.setTheme("ace/theme/monokai");
-
-        this.HTMLEditor.session.setMode("ace/mode/html");
-        this.CSSEditor.session.setMode("ace/mode/css");
-        this.JSEditor.session.setMode("ace/mode/javascript");
-
-        this.HTMLEditor.session.setUseWorker(false);
-        this.CSSEditor.session.setUseWorker(false);
-        this.JSEditor.session.setUseWorker(false);
+    async setupEditors(){
+        this.HTMLEditor = monaco.editor.create(document.getElementById('HTMLEditor'), {
+            theme: 'vs-dark',
+            language: 'html',
+            automaticLayout: true
+        });
+        this.CSSEditor = monaco.editor.create(document.getElementById('CSSEditor'), {
+            theme: 'vs-dark',
+            language: 'css',
+            automaticLayout: true
+        });
+        this.JSEditor = monaco.editor.create(document.getElementById('JSEditor'), {
+            theme: 'vs-dark',
+            language: 'FormItWSM',
+            automaticLayout: true
+        });
 
         const htmlContent = localStorage.getItem('currentHTMLValue') || defaultRepoValues.html;
         const cssContent = localStorage.getItem('currentCSSValue') || defaultRepoValues.css;
         const scriptContent = localStorage.getItem('currentJSValue') || defaultRepoValues.script;
 
-        this.HTMLEditor.setValue(htmlContent, 1);
-        this.CSSEditor.setValue(cssContent, 1);
-        this.JSEditor.setValue(scriptContent, 1);
-
-        const langTools = ace.require("ace/ext/language_tools");
-
-        const testCompleter ={
-            getCompletions: function(editor, session, pos, prefix, callback) {
-                var completions = [];
-                ["API1", "API2"].forEach(function(apiWord) {
-
-                    completions.push({
-                        value: apiWord,
-                        meta: "this is a test tip",
-
-                    });
+        this.HTMLEditor.setValue(htmlContent);
+        this.CSSEditor.setValue(cssContent);
+        this.JSEditor.setValue(scriptContent);
+
+        monaco.languages.register({ id: 'FormItWSM' });
+
+        //27, javascript
+        const javascriptLoader = await monaco.languages.getLanguages()[27].loader();
+
+        monaco.languages.setMonarchTokensProvider('FormItWSM', javascriptLoader.language);
+
+        const autocompleteURL = "https://formit3d.github.io/FormItExamplePlugins/docs/autocomplete.json";
+        fetch(autocompleteURL)
```

Comment:
Suggest you use async/await instead of promises. You're already using async await.